# Schätzung Parameter 1. Ordnung

In [4]:
import numpy as np 
import pandas as pd

from statsmodels.formula.api import ols
from patsy.contrasts import Treatment, Sum 
from statsmodels.stats.anova import anova_lm

In [5]:
# Daten
reakt = pd.DataFrame(
{'x1': np.array([120,160,120,160,140,140]), 
'x2': np.array([50,50,70,70,60,60]),
'y': np.array([52,62,60,70,63,65])
}
)

In [6]:
fit = ols('y ~ x1 + x2', data=reakt).fit() 
fit.params

Intercept    3.00
x1           0.25
x2           0.40
dtype: float64

Geschätztes Modell 1. Ordnung: $\hat{y} = 3 + 0.25x_1 + 0.4x_2$

# Schätzung Variablen 2. Ordnung
### mit Rotatable central composite design

In [7]:
import numpy as np 
import pandas as pd

from statsmodels.formula.api import ols
from patsy.contrasts import Treatment, Sum 
from statsmodels.stats.anova import anova_lm

In [8]:
reakt = pd.DataFrame(
{'Zeit': np.array([80,80,100,100,90,90,76,104,90]), 
'Temp': np.array([195,235,195,235,187,243,215,215,215]), 
'y': np.array([78,76,72,75,74,76,77,72,80])
})

In [9]:
fit = ols('y~Zeit+Temp+I(Temp*Temp)+I(Zeit*Zeit)+I(Zeit*Temp)',data=reakt).fit() 
fit.params

Intercept        -277.587423
Zeit                3.229716
Temp                2.024201
I(Temp * Temp)     -0.005960
I(Zeit * Zeit)     -0.026390
I(Zeit * Temp)      0.006250
dtype: float64

Wirkungsfläche 2. Ordnung

\begin{align*}
\hat{Y} &= \hat{\beta}_0 + \hat{\beta}_1 x_1 + \hat{\beta}_2 x_2 + \hat{\beta}_{11} x_1^2 + \hat{\beta}_{22} x_2^2 + \hat{\beta}_{12} x_1 x_2 \\
&= -277.6 + 2.02x_1 + 3.23x_2 - 0.00596x_1^2 - 0.0264x_2^2 + 0.00625x_1x_2
\end{align*}
